File: examples/PlayerManager.lua
local PlayerManager = {}
PlayerManager.__index = PlayerManager
function PlayerManager.new()
    local self = setmetatable({}, PlayerManager)
    self._players = game:GetService("Players")
    self._playerCount = 0
    return self
end
function PlayerManager:Initialize()
    print("PlayerManager initializing...")
    self._players.PlayerAdded:Connect(function(...)
        return self:OnPlayerAdded(...)
    end)
    self._players.PlayerRemoving:Connect(function(...)
        return self:OnPlayerRemoving(...)
    end)
end
function PlayerManager:OnPlayerAdded(player)
    self._playerCount = self._playerCount + 1
    print("Player joined: " .. player.Name)
    local character = player and player.Character
    local displayName = player.DisplayName or player.Name
    if character ~= nil then
        print(displayName .. " has a character")
    end
end
function PlayerManager:OnPlayerRemoving(player)
    self._playerCount = self._playerCount - 1
    print("Player left: " .. player.Name)
end
function PlayerManager.GetVersion()
    return "1.0.0"
end
return PlayerManager

File: examples/PlayerManager.cs
// PlayerManager.cs — roblox-cs class example
// Demonstrates: class emission, constructor, instance methods,
//               static methods, this -> self, events, null-conditional
using System;
using Roblox;
using Roblox.Services;
using Roblox.Instances;

public class PlayerManager
{
    private Players _players;
    private int _playerCount;

    public PlayerManager()
    {
        _players = Game.GetService<Players>();
        _playerCount = 0;
    }

    public void Initialize()
    {
        Console.WriteLine("PlayerManager initializing...");
        _players.PlayerAdded += this.OnPlayerAdded;
        _players.PlayerRemoving += this.OnPlayerRemoving;
    }

    private void OnPlayerAdded(Player player)
    {
        _playerCount = _playerCount + 1;
        Console.WriteLine("Player joined: " + player.Name);

        var character = player?.Character;
        var displayName = player.DisplayName ?? player.Name;

        if (character != null)
        {
            Console.WriteLine(displayName + " has a character");
        }
    }

    private void OnPlayerRemoving(Player player)
    {
        _playerCount = _playerCount - 1;
        Console.WriteLine("Player left: " + player.Name);
    }

    public static string GetVersion()
    {
        return "1.0.0";
    }
}
